src/osd/ECBackend.cc

```diff
@@ -1730,10 +1739,25 @@ void ECBackend::do_read_op(ReadOp &op)
       msg->trace.init("ec sub read", nullptr, &op.trace);
       msg->trace.keyval("shard", i->first.shard.id);
     }
-    get_parent()->send_message_osd_cluster(
-      i->first.osd,
-      msg,
-      get_parent()->get_epoch());
+    if (i->first == get_parent()->whoami_shard()) {
+      should_read_local = true;
+      local_read_op = i->second;
```

Comment:
it looks like we're still constructing hte MOSDECSubOpRead for ourselves.. and then leaking it.  I think this condition should be at the top of the loop?